code/game/objects/items/weapons/storage/fancy.dm

```diff
@@ -336,17 +336,31 @@
 /obj/item/weapon/storage/fancy/cigarettes/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
 	if(!istype(M, /mob))
 		return
-
-	if(equip_from_box && M == user && user.zone_sel.selecting == "mouth" && contents.len > 0 && !user.wear_mask)
-		var/obj/item/clothing/mask/cigarette/W = new /obj/item/clothing/mask/cigarette(user)
-		reagents.trans_to(W, (reagents.total_volume/contents.len))
-		user.equip_to_slot_if_possible(W, slot_wear_mask)
-		reagents.maximum_volume = 15 * contents.len
-		contents.len--
+	var/list/cigs = list()
+	for(var/obj/item/clothing/mask/cigarette/cig in contents)
+		cigs.Add(cig)
+	if (user.a_intent != I_HELP || user.zone_sel.selecting != "mouth")
+		return ..()
+	if(!cigs.len)
+		to_chat(user, "<span class='notice'>There are no cigarettes left in the pack.</span>")
+		return
+	var/obj/item/clothing/mask/cigarette/mycig = cigs[cigs.len]
+	if(equip_from_box && M == user && !user.wear_mask)
+		mycig.forceMove(user)
+		user.equip_to_slot_if_possible(mycig, slot_wear_mask)
```

Comment:
Like the name implies, this can fail, so you might want to check for the return value and abort.